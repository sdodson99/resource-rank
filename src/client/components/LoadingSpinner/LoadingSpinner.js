import React from 'react';
import PropTypes from 'prop-types';
import styles from './LoadingSpinner.module.css';

const LoadingSpinner = ({ height, width }) => (
  <div className={styles.LoadingSpinner} data-testid="LoadingSpinner">
    <img
      className="animate-spin align-middle"
      src="/img/spinner.svg"
      height={height}
      width={width}
    />
  </div>
);

LoadingSpinner.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
};

LoadingSpinner.defaultProps = {
  height: 50,
  width: 50,
};

export default LoadingSpinner;
