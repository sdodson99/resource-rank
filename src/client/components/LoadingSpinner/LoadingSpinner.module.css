.LoadingSpinner {
  @apply flex justify-center animate-spin align-middle;
}
