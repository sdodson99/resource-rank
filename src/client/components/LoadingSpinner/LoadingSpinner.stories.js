import React from 'react';
import LoadingSpinner from './LoadingSpinner';

export default {
  component: LoadingSpinner,
  title: 'Components/LoadingSpinner',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <LoadingSpinner {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Size = Template.bind({});
Size.args = {
  size: 300,
};

export const Color = Template.bind({});
Color.args = {
  size: 100,
  color: 'red',
};
