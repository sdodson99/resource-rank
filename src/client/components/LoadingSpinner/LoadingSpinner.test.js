import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LoadingSpinner from './LoadingSpinner';
import renderer from 'react-test-renderer';

describe('<LoadingSpinner />', () => {
  test('it should mount', () => {
    render(<LoadingSpinner />);

    const loadingSpinner = screen.getByTestId('LoadingSpinner');

    expect(loadingSpinner).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<LoadingSpinner size={100} color={'red'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
