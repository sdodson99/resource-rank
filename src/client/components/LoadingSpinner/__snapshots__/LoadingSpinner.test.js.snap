// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> should render correctly 1`] = `
<div
  className="LoadingSpinner"
  data-testid="LoadingSpinner"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": 50,
        "overflow": "hidden",
        "position": "relative",
        "width": 50,
      }
    }
  >
    <noscript>
      <img
        className="animate-spin"
        decoding="async"
        src="/img/spinner.svg"
        srcSet="/_next/image?url=%2Fimg%2Fspinner.svg&w=64&q=75 1x, /_next/image?url=%2Fimg%2Fspinner.svg&w=128&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
    <img
      className="animate-spin"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
